app/views/admin/users/index.slim

```diff
@@ -42,11 +42,11 @@ header.main-content__header role="banner"
       )
 
   div style="margin-left: 10px"
-    = link_to(                                                                             \
+    = link_to(                                                                              \
         Settings.registrations_enabled? ? "Disable registrations" : "Enable registrations", \
-        admin_users_toggle_registrations_path,                                             \
-        class: "button",                                                                   \
-        method: :post                                                                      \
+        admin_users_set_registrations_path(value: !Settings.registrations_enabled?),        \
```

Comment:
If you think that's better, I'll change it.

Reply:
But the user_controller is very polluted. I'd prefer to do that when we clean that up.

Reply:
It's one variable, it won't change that much. I find it much more serious the view accessing the database and having internal knowledge instead of focusing on displaying stuff.

Reply:
It's pretty simple, the `users_controller` is already 100 lines long, if I had that it will go over that limit and I will refactor a bunch of stuff. My suggestion: remove those unused buttons from the `users_controller` and view.

Reply:
Then you have a different issue: refactor the `users_controller`. That doesn't mean you should break contexts and concerns and start accessing the database in the view.